### ISSUE TITLE
Issue on X11 fedora 34.

### ISSUE THREAD

**Issue by basta** [patriciogonzalezvivo/glslViewer#214]
Something I faced and solved. 
When compiling according to instructions I ended up with glfw that expected wayland, when calling glslViewer i got an error like:
`glfw: Wayland: No display detected,` or something similiar. 
I needed to compile glfw manually and then recompile glslViewer.
Not sure what to do with this info, but I thought it might help other fedora people.

The problem _might_ have been with glslViewer. Because glxgears worked and I think that uses glfw? Not sure

**Comment by patriciogonzalezvivo** [comment]
@basta thanks so much for share this findings! Do you mind making a note or updating the documentation on the wiki?

**Comment by basta** [comment]
Sure thing, I am just worried that it's not a common problem. Not sure if I have some weird configuration.

Also, I have to add, since I have you here: Thanks so much for all your work, on the Book of Shaders and everything, you made me find a new passion for shaders

**Comment by patriciogonzalezvivo** [comment]
Oh man! thanks so much! your words means a lot

Good point about this maybe a edge case. Let's live it here. If it pops again for more people we can write something. I don't use fedora but wayland slowly becoming the standard could bring more of this kind of issues

**Comment by basta** [comment]
Very well then. Let's wait. I think that google will lead people here anyway, if they google the error

**Comment by patriciogonzalezvivo** [comment]
Hi @basta! I'm doing some housekeeping and moving things around a bit for 1.7.0 release. One of the big steps is adding LIBAV support by default in Linux and MacOS. For that ffmpeg and libav dev packages need to install as well. Do you mind cloning `main` branch (I just switch it as the default), and see if the wiki instractions [here](https://github.com/patriciogonzalezvivo/glslViewer/wiki/Compiling#b-compiling-on-fedora-distributions-with-glfw) are to date?

**Comment by basta** [comment]
Seems ok. Downloaded the dependencies and compiled the main branch and all seems to work. Haven't tested any special features though. 
Shaders can output audio as well right? Maybe I should test that as Fedora now uses Pipewire instead of Pulseaudio. If you think that testing that is useful and could point me to a shader that produces audio I can test that as well.

**Comment by patriciogonzalezvivo** [comment]
No, the othe way around. Audio stream gets converted into a texture, to do audio reactive shaders https://github.com/patriciogonzalezvivo/glslViewer/tree/main/examples/2D/06_audio

Thanks for checking!!

**Comment by rivenirvana** [comment]
Just encountered this as well and looked into it. I believe this is because the specfile that builds the RPM for glfw in the official Fedora repos has `-DGLFW_USE_WAYLAND=ON` as a cmake build option, because of Wayland being default now. That means there is no other workaround for X11 Fedora users other than to compile glfw manually with the correct build configuration, or if someone packages a different RPM of glfw for X11.

[glfw specfile used by Fedora, line 77](https://src.fedoraproject.org/rpms/glfw/blob/rawhide/f/glfw.spec#_77)
[Relevant commit](https://src.fedoraproject.org/rpms/glfw/c/9765052882702457a1c1f85e9c44602f3cf4158b?branch=rawhide)
